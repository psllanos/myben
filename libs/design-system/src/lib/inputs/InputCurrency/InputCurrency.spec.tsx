import { fireEvent, render, screen } from '@testing-library/react'
import { useState } from 'react'
import { InputCurrency } from '..'

function ControlledInputCurrency(props: { [key: string]: unknown }): JSX.Element {
    const [value, setValue] = useState<number | null>(20)
    return <InputCurrency value={value} onChange={setValue} {...props} />
}

describe('InputCurrency', () => {
    describe('when rendered', () => {
        it('should display a dollar sign by default', () => {
            const component = render(<InputCurrency value={null} onChange={() => null} />)

            expect(screen.getByText('$')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
        it('should display a specified currency symbol', () => {
            const component = render(
                <InputCurrency symbol="symbol" value={null} onChange={() => null} />
            )

            expect(screen.getByText('symbol')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
    })

    describe('when interacted with', () => {
        it('should call `onChange` when the entered number changes', () => {
            const onChangeMock = jest.fn()

            render(<ControlledInputCurrency data-testid="input" onChange={onChangeMock} />)
            const input = screen.getByTestId('input')

            fireEvent.focus(input)
            fireEvent.change(input, { target: { value: -123 } })
            expect(onChangeMock).toHaveBeenCalledWith(123) // Negative inputs not accepted, will be converted to positive
        })
    })
})
