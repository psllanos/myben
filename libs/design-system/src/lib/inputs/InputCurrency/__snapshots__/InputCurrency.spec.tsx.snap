// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCurrency when rendered should display a dollar sign by default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="flex w-full flex-col"
      >
        <div
          class="flex h-10 text-white rounded border overflow-hidden relative border-gray-700 focus-within:ring focus-within:ring-opacity-10 focus-within:border-cyan focus-within:ring-cyan"
        >
          <span
            class="absolute left-3 top-1/2 -translate-y-1/2 flex items-center justify-center text-gray-50 text-base select-none bg-gray-500"
          >
            $
          </span>
          <input
            class="min-w-0 py-0 w-full text-base font-light leading-none border-0 focus:outline-none focus:ring-0 placeholder-gray-100 disabled:placeholder-gray-200 disabled:text-gray-100 bg-gray-500 pl-8 pr-3"
            inputmode="numeric"
            type="text"
            value=""
          />
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="flex w-full flex-col"
    >
      <div
        class="flex h-10 text-white rounded border overflow-hidden relative border-gray-700 focus-within:ring focus-within:ring-opacity-10 focus-within:border-cyan focus-within:ring-cyan"
      >
        <span
          class="absolute left-3 top-1/2 -translate-y-1/2 flex items-center justify-center text-gray-50 text-base select-none bg-gray-500"
        >
          $
        </span>
        <input
          class="min-w-0 py-0 w-full text-base font-light leading-none border-0 focus:outline-none focus:ring-0 placeholder-gray-100 disabled:placeholder-gray-200 disabled:text-gray-100 bg-gray-500 pl-8 pr-3"
          inputmode="numeric"
          type="text"
          value=""
        />
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`InputCurrency when rendered should display a specified currency symbol 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="flex w-full flex-col"
      >
        <div
          class="flex h-10 text-white rounded border overflow-hidden relative border-gray-700 focus-within:ring focus-within:ring-opacity-10 focus-within:border-cyan focus-within:ring-cyan"
        >
          <span
            class="absolute left-3 top-1/2 -translate-y-1/2 flex items-center justify-center text-gray-50 text-base select-none bg-gray-500"
          >
            symbol
          </span>
          <input
            class="min-w-0 py-0 w-full text-base font-light leading-none border-0 focus:outline-none focus:ring-0 placeholder-gray-100 disabled:placeholder-gray-200 disabled:text-gray-100 bg-gray-500 pl-8 pr-3"
            inputmode="numeric"
            type="text"
            value=""
          />
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="flex w-full flex-col"
    >
      <div
        class="flex h-10 text-white rounded border overflow-hidden relative border-gray-700 focus-within:ring focus-within:ring-opacity-10 focus-within:border-cyan focus-within:ring-cyan"
      >
        <span
          class="absolute left-3 top-1/2 -translate-y-1/2 flex items-center justify-center text-gray-50 text-base select-none bg-gray-500"
        >
          symbol
        </span>
        <input
          class="min-w-0 py-0 w-full text-base font-light leading-none border-0 focus:outline-none focus:ring-0 placeholder-gray-100 disabled:placeholder-gray-200 disabled:text-gray-100 bg-gray-500 pl-8 pr-3"
          inputmode="numeric"
          type="text"
          value=""
        />
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
